runtime/src/dispatcher.rs

```diff
@@ -253,6 +268,10 @@ impl Dispatcher {
                         signed_policy_raw,
                     );
                 }
+                Ok((_ctx, _id, Body::RuntimeAbortRequest {})) => {
+                    info!(self.logger, "Received abort request");
+                    break 'dispatch;
```

Comment:
Hm, won't terminating the dispatcher bring the runtime down? The abort should be handled like the following:
* Abort is considered finished once the current request that is being handled by the dispatcher is completed (e.g., after this big `match` is done processing whatever it is currently processing). It shouldn't interrupt the dispatcher as then you need to set it up again and this probably leads to restarting the runtime? Instead, the loop should just continue here with the next `rx.recv()` (after signalling that an abort request has been processed).
* It would be cool (read: optional) if the default transaction dispatcher (`MethodDispatcher` under `runtime/src/transaction/dispatcher.rs`) could abort while in the middle of processing a batch, (e.g. during `dispatch_batch`). This could be done by just having a synchronized flag (e.g., an `AtomicBool`) that you could check to see if you need to stop processing the batch and return immediately with an error.

This second part is optional and the whole thing would be useful even if it is not implemented.

Reply:
Ah, i thought we actually want to restart the runtime, but yeah there is no reason for it actually so it makes more sense avoiding that and just waiting for the currently processed batch to finish.

Yeah can also do the 2nd part as part of this PR :+1: